File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int n=numbers.length;
        int i=0,j=n-1;
        while(i<n && j>0 && i<j){
            if(numbers[i]+numbers[j]>target) j--;
            else if(numbers[i]+numbers[j]<target) i++;
            else return new int[]{i+1,j+1};
        }
        return new int[2];
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        exist = {}
        for i,n in enumerate(nums):
            if target - n in exist:
                return [exist[target-n], i]
            else:
                exist[n]=i
        return [-1,-1]
        